src/Workleap.Extensions.Mongo.Analyzers/IndexAttributeUsageAnalyzer.cs

```diff
@@ -78,16 +78,35 @@ public void AnalyzeOperationInvocation(OperationAnalysisContext context)
                 return;
             }
 
-            var hasIndexAttribute = containingClassSymbol.GetAttributes()
-                .Any(x => x.AttributeClass != null && this._mongoIndexAttributes.Contains(x.AttributeClass));
-
+            var hasIndexAttribute = this.IsMethodOrClassContainingIndexAttribute(context);
             if (!hasIndexAttribute)
             {
                 context.ReportDiagnostic(UseIndexAttributeRule, operation);
-                this._symbolsWithoutAttributes.TryAdd(containingClassSymbol, true);
+                this._symbolsWithoutAttributes.TryAdd(containingMethodSymbol, true);
```

Comment:
`_symbolsWithoutAttributes` is supposed to allow the analyzer to early exit if an attribute isn't needed because we already analyzed the type / method before.

I feel like this scenario will be not optimized:
1. I have a class MyHandler that has 3 methods
2. Each method uses a mongo collection
3. I have a index attribute at the class level

Because there's an attribute at the class level, I would expect the analyzer to early exit. But here I feel like because the `_symbolsWithoutAttributes` now only store information at the method level, we would still analyze each method invocation instead of early exiting.

Reply:
I dit a test and with the inverse logic on the Provisionning project we go from 19ms to 2ms

![image](https://github.com/gsoft-inc/wl-extensions-mongo/assets/8277266/ed13e40b-228c-49ee-b807-9a3611f63876)
![image](https://github.com/gsoft-inc/wl-extensions-mongo/assets/8277266/20b1c623-f8dc-43b7-a5bf-633848619ff3)

Reply:
I did the change, tell me what you think :)